arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -0,0 +1,265 @@
+        syncGeodatabaseParameters.geodatabaseSyncDirection = AGSSyncDirection.bidirectional
+        syncGeodatabaseParameters.rollbackOnFailure = false
+        
+        // Specify the layer IDs of the feature tables to sync.
+        for geodatabaseFeatureTable in geodatabase.geodatabaseFeatureTables {
+            let serviceLayerId = geodatabaseFeatureTable.serviceLayerID
+            let syncLayerOption = AGSSyncLayerOption(layerID: serviceLayerId, syncDirection: .bidirectional)
+            syncGeodatabaseParameters.layerOptions.append(syncLayerOption)
+        }
+        // Create a sync job with the parameters and start it.
+        let syncGeodatabaseJob = geodatabaseSyncTask.syncJob(with: syncGeodatabaseParameters, geodatabase: geodatabase)
+        self.syncJob = syncGeodatabaseJob
+        syncGeodatabaseJob.start(statusHandler: { (status: AGSJobStatus) in
+            SVProgressHUD.show(withStatus: status.statusString())
+        },
+                                 completion: { (result: [AGSSyncLayerResult]?, error: Error?) in
+                                    SVProgressHUD.dismiss()
+                                    if let error = error {
+                                        self.presentAlert(error: error)
+                                    } else {
+                                            self.presentAlert(title: "Geodatabase sync sucessful")
+                                            self.syncButton.isEnabled = false
+                                            self.syncButton.isHidden = true
+                                            self.tapFeatureLabel.isHidden = false
+                                        }
+                                    }
+        )
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["EditAndSyncFeaturesViewController"]
+    }
+}
+
+// Allows the user to interactively select and move features on the map.
+extension EditAndSyncFeaturesViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Move the selected feature to the tapped location and update it in the feature table.
+        if self.selectedFeature != nil {
+            let point = mapView.screen(toLocation: screenPoint)
+            if AGSGeometryEngine.geometry(point, intersects: areaOfInterest) {
+                selectedFeature.geometry = point
+                selectedFeature.featureTable?.update(selectedFeature) { [weak self] (error: Error?) in
+                    if let error = error {
+                        self!.presentAlert(error: error)
+                    } else {
+                        self!.syncButton.isHidden = false
+                        self!.syncButton.isEnabled = true
```

Comment:
In general, repeatedly accessing a `weak` value is not a good idea since the value could become `nil` in between accesses to it. We know that can't happen here, but `self` is also being force unwrapped multiple times. Instead, please either unwrap `self` at the top of the closure or extract the contents of the closure into a method so that `self` only needs to be referenced once (to call the method).